src/main/java/com/coreos/jetcd/internal/impl/KVImpl.java

```diff
@@ -23,14 +23,19 @@
 import com.coreos.jetcd.options.GetOption;
 import com.coreos.jetcd.options.PutOption;
 import java.util.concurrent.CompletableFuture;
+import javax.annotation.concurrent.GuardedBy;
 
 /**
  * Implementation of etcd kv client.
  */
 class KVImpl implements KV {
 
   private final ClientConnectionManager connectionManager;
-  private final KVGrpc.KVFutureStub stub;
+
+  // stubLock protects stub
+  private final Object stubLock = new Object();
+  @GuardedBy("this")
```

Comment:
If you use an external lock object this annotation should reference it, like ``@GuardedBy("stubLock")``